File: csharp/src/Ice/LocalException.cs
// Copyright (c) ZeroC, Inc.

#nullable enable

namespace Ice;

/// <summary>
/// Base class for Ice run-time exceptions.
/// </summary>
public class LocalException : Ice.Exception
{
    /// <summary>
    /// Constructs a LocalException.
    /// </summary>
    /// <param name="message">The exception message.</param>
    /// <param name="innerException">The inner exception.</param>
    public LocalException(string? message, System.Exception? innerException = null)
        : base(message, innerException)
    {
    }

    public override string ice_id() => "::Ice::LocalException";
}

File: java/src/Ice/src/main/java/com/zeroc/Ice/LocalException.java
// Copyright (c) ZeroC, Inc.

package com.zeroc.Ice;

/** Base class for Ice run-time exceptions. */
public class LocalException extends Exception {
  public LocalException(String message) {
    super(message);
  }

  public LocalException(String message, Throwable cause) {
    super(message, cause);
  }

  @Override
  public String ice_id() {
    return "::Ice::LocalException";
  }

  private static final long serialVersionUID = 0L;
}
